app/src/main/java/org/dhis2/Bindings/TrackedEntityInstanceExtensions.kt

```diff
@@ -25,3 +28,44 @@ fun MutableList<TrackedEntityInstance>.filterDeletedEnrollment(
 
     return this
 }
+
+fun MutableList<TrackedEntityInstance>.filterEvents(
+    d2: D2,
+    eventPeriods: MutableList<DatePeriod>,
+    program: String?
+): MutableList<TrackedEntityInstance> {
```

Comment:
I miss the unit test. Also, I think it's easier to use "filter" from kotlin

Reply:
You can't use the filter functions as they create new lists. It has to be done with the iterator.

Reply:
unit test done